src/inset_state/flatten_ellipse_density.cpp

```diff
@@ -0,0 +1,488 @@
+#include "inset_state.h"
+
+void InsetState::min_ellipses()
+{
+  for (auto &gd : geo_divs_) {
+    gd.clear_min_ellipses();
+    for (auto &pwh : gd.polygons_with_holes()) {
+      auto ext_ring_actual = pwh.outer_boundary();
+      std::vector<double> x_coords, y_coords;
+      for (const auto &pt : ext_ring_actual) {
+        x_coords.push_back(pt.x());
+        y_coords.push_back(pt.y());
+      }
+      std::nth_element(
+        x_coords.begin(),
+        x_coords.begin() + x_coords.size() / 2,
+        x_coords.end());
+      std::nth_element(
+        y_coords.begin(),
+        y_coords.begin() + y_coords.size() / 2,
```

Comment:
Clang-Tidy: Narrowing conversion from 'std::vector<double>::size_type' (aka 'unsigned long') to signed type 'std::__wrap_iter<double *>::difference_type' (aka 'long') is implementation-defined